ml-kem/src/kem.rs

```diff
@@ -253,6 +266,190 @@ where
     }
 }
 
+/// The serialization of the private key is a choice between three different formats
+/// [according to PKCS#8](https://lamps-wg.github.io/kyber-certificates/draft-ietf-lamps-kyber-certificates.html#name-private-key-format).
+///
+/// “For ML-KEM private keys, the privateKey field in `OneAsymmetricKey`
+/// contains one of the following DER-encoded `CHOICE` structures.
+/// The seed format is a fixed 64-byte `OCTET STRING` (66 bytes total
+/// with the 0x8040 tag and length) for all security levels,
+/// while the expandedKey and both formats vary in size by security level”
+#[cfg(feature = "pkcs8")]
+#[derive(Clone, Debug, pkcs8::der::Choice)]
+pub enum PrivateKeyChoice<'o> {
+    /// FIPS 203 format for an ML-KEM private key: a 64-octet seed
+    #[asn1(tag_mode = "IMPLICIT", context_specific = "0")]
+    Seed(OctetStringRef<'o>),
+    /// FIPS 203 format for an ML-KEM private key: the decapsulation key resulting from PKE's `KeyGen` operation
+    Expanded(OctetStringRef<'o>),
+    /// In this setting both key formats are provided in a `PrivateKeyBothChoice` `struct`
+    Both(PrivateKeyBothChoice<'o>),
+}
+
+/// The private key's `Both` variant contains the seed as well as the expanded key.
+#[cfg(feature = "pkcs8")]
+#[derive(Clone, Debug, pkcs8::der::Sequence)]
+pub struct PrivateKeyBothChoice<'o> {
+    /// FIPS 203 format for an ML-KEM private key: a 64-octet seed
+    pub seed: OctetStringRef<'o>,
+    /// FIPS 203 format for an ML-KEM private key: the decapsulation key resulting from PKE's `KeyGen` operation
+    pub expanded: OctetStringRef<'o>,
+}
```

Comment:
Not sure how I feel about supporting all of these. IMO this approach is something of a mistake which has been widely criticized and exists only because a handful of HSM vendors shipped draft standards too soon and refuse to change.